velox/connectors/hive/HiveConnector.cpp

```diff
@@ -81,28 +91,67 @@ std::string HiveTableHandle::toString() const {
 
 HiveDataSink::HiveDataSink(
     std::shared_ptr<const RowType> inputType,
-    const std::string& filePath,
-    velox::memory::MemoryPool* memoryPool)
-    : inputType_(inputType) {
+    std::shared_ptr<const HiveInsertTableHandle> insertTableHandle,
+    const ConnectorQueryCtx* FOLLY_NONNULL connectorQueryCtx,
+    std::shared_ptr<WriteProtocol> writeProtocol)
+    : inputType_(inputType),
+      insertTableHandle_(insertTableHandle),
```

Comment:
std::move here and everywhere else where shared_ptr is used

Reply:
Move in a separate PR due to dependencies, https://github.com/facebookincubator/velox/pull/2832, https://github.com/prestodb/presto/pull/18492.